backend/devflow-ai/src/main/kotlin/com/devflow/ai/infrastructure/PromptTemplateRepositoryImpl.kt

```
@@ -0,0 +1,64 @@
+    }
+    
+    override fun save(promptTemplate: PromptTemplate): PromptTemplate {
+        val document = PromptTemplateDocument.fromDomain(promptTemplate)
+        val saved = mongoRepository.save(document)
+        return saved.toDomain()
+    }
+    
+    override fun deleteById(id: String) {
+        try {
+            mongoRepository.deleteById(ObjectId(id))
+        } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)